### PR TITLE
fdc3 e2e test

### PR DESCRIPTION
**Pull request by agyen**
Automated test for displaying pricing data and opening chart using sikuli

### CODE REVIEW

**Comment by ZKRobi** [comment]
Could you add a readme explaining prerequisites and instructions on running the tests?

File: README.md [hunk 1]
```diff
@@ -150,3 +150,19 @@ The necessary solutions have to be built in the following order:
 1. Open the Shell Solution
 2. Choose "Shell" as the startup project
 3. Run
+
+### Running FDC3 sikuli e2e tests
+#### Prerequisite
+1. Java version 8 or later
+
+#### IDE setup
+1. [Download](https://launchpad.net/sikuli/+download) sikuli IDE jar file.
```
Comment (ZKRobi) [inline README.md:159]: I don't think this is the official link, can we look into this?

File: README.md [hunk 2]
```diff
@@ -150,3 +150,19 @@ The necessary solutions have to be built in the following order:
...
+2. [Download](https://repo1.maven.org/maven2/org/python/jython-standalone/2.7.1/jython-standalone-2.7.1.jar) jython standalone jar file.
```
Comment (ZKRobi) [inline README.md:160]: As far as I can tell this is not necessary for the latest version

File: README.md [hunk 3]
```diff
@@ -150,3 +150,19 @@ The necessary solutions have to be built in the following order:
...
+1. [Download](https://launchpad.net/sikuli/+download) sikuli IDE jar file.
+2. [Download](https://repo1.maven.org/maven2/org/python/jython-standalone/2.7.1/jython-standalone-2.7.1.jar) jython standalone jar file. 
+3. Have Sikuli IDE jar file and the jython standalone jar file in the same directory.
+4. Launch the IDE with a double click on the sikuli IDE jar file
+
+#### Run tests
+1. Launch FDC3-Chart-and-Grid
+2. On the Sikuli IDE click on file -> open
+3. Locate the test file and select
+4. Click on run(Make sure FDC3-Chart-and-Grid is running in the background and the next application after the IDE)
```
Comment (ZKRobi) [inline README.md:168]: Based on the attached screenshots the "Shell" launch configuration needs to be started (but yes, the hosting is necessary).
Please clarify this a bit as there is a Chart and Grid launch configuration as well that looks different and wouldn't work.

**Review by ZKRobi — CHANGES_REQUESTED** [review]

File: examples/fdc3-chart-and-grid/fdc3-chart-and-grid.sikuli/grid-and-chat-display-automated-test.py [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+    chart = wait("./screenshots/open-chart-button.png", 20)
+    click(chart)
+    if exists("./screenshots/chart-opened.png"):
+        click(Pattern("./screenshots/goog-symbol.png").exact())
+        wait(10)
+    else:
+        popup("Chart not available to proceed")
+        exit()
+else:
+    popup("Module not found, can't continue")
```
Comment (ZKRobi) [inline examples/fdc3-chart-and-grid/fdc3-chart-and-grid.sikuli/grid-and-chat-display-automated-test.py:12]: Please clarify this message to reflect that it's looking for the folder :)

File: README.md (hunk 1 above)
Comment (ZKRobi) [reply README.md:159]: Ok now I can see launchpad is official as well, strange project

File: examples/fdc3-chart-and-grid/fdc3-chart-and-grid.sikuli/grid-and-chat-display-automated-test.py [hunk 5]
```diff
@@ -0,0 +1,13 @@
+if exists("./screenshots/modules"):
```
Comment (ZKRobi) [inline examples/fdc3-chart-and-grid/fdc3-chart-and-grid.sikuli/grid-and-chat-display-automated-test.py:1]: Quite unfortunately this is relative to the working directory, and not the script. We can live with this, or solve it later, but it's worth noting in the readme

**Review by ZKRobi — DISMISSED** [review]

**Review by lilla28 — DISMISSED** [review]

**Review by ZKRobi — APPROVED** [review]

**Pull request merged**